python/lsst/daf/butler/_butler.py

```diff
@@ -333,10 +333,13 @@ def makeRepo(root: str, config: Union[Config, str, None] = None, standalone: boo
             if not os.path.isdir(uri.ospath):
                 safeMakeDir(uri.ospath)
         elif uri.scheme == "s3":
-            s3 = boto3.resource("s3")
-            # implies bucket exists, if not another level of checks
-            bucket = s3.Bucket(uri.netloc)
-            bucket.put_object(Bucket=uri.netloc, Key=uri.relativeToPathRoot)
+            # implies bucket already exists
+            s3 = boto3.client("s3")
+            if uri.relativeToPathRoot != ".":
```

Comment:
Maybe say that this test is to see whether the gen3 repo represents the root of the bucket.